### DISCUSSION TITLE
$bot->message()->forward_from->id ErrorException

### DISCUSSION THREAD

**Discussion by Jorge-S13** [nutgram/nutgram#473] [Q&A]
I have the following error. I need to get $bot->message()->forward_from->id. This is a feedback bot. The user enters their feedback to the bot, the bot sends it to the group. Next, the moderator replies to the message in the group using the reply method. The moderator's response is sent back to the user by the bot.
Here is my code
```php
$bot = new Nutgram($_ENV['TELEGRAM_TOKEN'],[
    'timeout' => $_ENV['CONNECT_TIMEOUT'],
    'logger' => ConsoleLogger::class,
]);

$bot->onCommand('start', function (Nutgram $bot) {
    $bot->sendMessage('Hello');
})->description('Start Command');

$bot->onCommand('feedBack', function (Nutgram $bot) {
    FeedBackConversation::begin($bot);
})->description('Feed Back Command');

$bot->onMessage(function (Nutgram $bot) {
    $message = $bot->message();

    if ($message->reply_to_message !== null && $message->chat->id === -855697254){
        $bot->onMessage(function (Nutgram $bot) {
            $forwardMessage = $bot->message();
                $forwardFrom = $forwardMessage->forward_from->id;
                $messageText = $forwardMessage->text;
                $bot->sendMessage($messageText,['chat_id' => $forwardFrom]);

        });
    }
});

$bot->run();
```
Here is my answer from telegram
```bash
[2023-05-25 12:05:38] DEBUG: getUpdates
{"ok":true,"result":[{"update_id":776724918,
"message":{"message_id":393,"from":{"id":3182829,"is_bot":false,"first_name":"Jorge","username":"sviridov13","language_code":"ru"},"chat":{"id":-855697254,"title":"Feed Backs Group","t
ype":"group","all_members_are_administrators":true},"date":1685016377,"reply_to_message":{"message_id":392,"from":{"id":5948028649,"is_bot":true,"first_name":"Opinionator\ud83d\udc7e",
"username":"Opinionatorbot"},"chat":{"id":-855697254,"title":"Feed Backs Group","type":"group","all_members_are_administrators":true},"date":1685016371,"forward_from":{"id":3182829,"is_bot":false,"first_name":"Jorge","username":"sviridov13","language_code":"ru"},"forward_date":1685016371,"text":"TEST"},"text":"TEST"}}]}
{"parameters":{"offset":776724918,"limit":100,"timeout":"10"},"options":{"timeout":11}}
```
And this error ErrorException: Attempt to read property "id" on null.
How can i implement it? And another question is how can I get and save the id of the groups to which I add my bot?

**Comment by Lukasss93** [comment]
1. You cannot use `$bot->onMessage` inside another `$bot->onMessage`. You must utilize a conversation instead.
2. Take a look at this page: https://nutgram.dev/docs/usage/conversations to understand the functioning of a conversation.
3. For an example of a feedback logic, refer to this [page](https://github.com/Lukasss93/telegram-stickeroptimizer/blob/master/app/Telegram/Conversations/FeedbackConversation.php).

**Reply by Jorge-S13** [reply]
> 1. You cannot use `$bot->onMessage` inside another `$bot->onMessage`. You must utilize a conversation instead.
> 2. Take a look at this page: https://nutgram.dev/docs/usage/conversations to understand the functioning of a conversation.
> 3. For an example of a feedback logic, refer to this [page](https://github.com/Lukasss93/telegram-stickeroptimizer/blob/master/app/Telegram/Conversations/FeedbackConversation.php).

how can i get the forward_from field? I get the next update.
`"message":{"message_id":437,"from":{"id":3182829,"is_bot":false,"first_name":"Jorge","username":"sviridov13","language_code":"ru"},"chat":{"id":-855697254,"title":"Feed Backs Group","t
ype":"group","all_members_are_administrators":true},"date":1685618296,"reply_to_message":{"message_id":429,"from":{"id":5948028649,"is_bot":true,"first_name":"Opinionator\ud83d\udc7e",
"username":"Opinionatorbot"},"chat":{"id":-855697254,"title":"Feed Backs Group","type":"group","all_members_are_administrators":true},"date":1685617481,"forward_from":{"id":3182829,"is_bot":false,"first_name":"Jorge","username":"sviridov13","language_code":"ru"},"forward_date":1685617480,"text":"sadfasdfasdf"},"text":"lelle"}}]}
{"parameters":{"offset":776724948,"limit":100,"timeout":"10"},"options":{"timeout":11}}
`
I want to get the following field.
`"forward_from":{"id":3182829,`
to further send a message to this user
if i do the following code `$this->forwardFrom = $bot->message()->forward_from->id; `then I get this error `Attempt to read property "id" ` the rest of the code works fine.

**Reply by Jorge-S13** [reply]
> > 1. You cannot use `$bot->onMessage` inside another `$bot->onMessage`. You must utilize a conversation instead.
> > 2. Take a look at this page: https://nutgram.dev/docs/usage/conversations to understand the functioning of a conversation.
> > 3. For an example of a feedback logic, refer to this [page](https://github.com/Lukasss93/telegram-stickeroptimizer/blob/master/app/Telegram/Conversations/FeedbackConversation.php).
> 
> how can i get the forward_from field? I get the next update. `"message":{"message_id":437,"from":{"id":3182829,"is_bot":false,"first_name":"Jorge","username":"sviridov13","language_code":"ru"},"chat":{"id":-855697254,"title":"Feed Backs Group","t ype":"group","all_members_are_administrators":true},"date":1685618296,"reply_to_message":{"message_id":429,"from":{"id":5948028649,"is_bot":true,"first_name":"Opinionator\ud83d\udc7e", "username":"Opinionatorbot"},"chat":{"id":-855697254,"title":"Feed Backs Group","type":"group","all_members_are_administrators":true},"date":1685617481,"forward_from":{"id":3182829,"is_bot":false,"first_name":"Jorge","username":"sviridov13","language_code":"ru"},"forward_date":1685617480,"text":"sadfasdfasdf"},"text":"lelle"}}]} {"parameters":{"offset":776724948,"limit":100,"timeout":"10"},"options":{"timeout":11}} ` I want to get the following field. `"forward_from":{"id":3182829,` to further send a message to this user if i do the following code `$this->forwardFrom = $bot->message()->forward_from->id; `then I get this error `Attempt to read property "id" ` the rest of the code works fine.

I solved my problem with `$this->forwardFrom = $bot->message()->reply_to_message->forward_from->id;`. Everything works correctly. Thank you
